<h1>Data preparation</h1>

This notebook has some data preparation (change column type and NaNs treatment in numerical and categorical variables)

In [21]:
import pandas as pd
import pickle
from ipynb.fs.full.funPyModeling import status, freq_tbl

<h3>Data loading</h3>

In [2]:
data = pd.read_csv("data/eph2.txt", sep = ",")

<h3>Dataset status</h3>

In [4]:
status(data)

variable  q_nan     p_nan  q_zeros  p_zeros  unique     type
0                 edad    134  0.038111        0      0.0      70  float64
1                 sexo      0  0.000000        0      0.0       2   object
2             alfabeto      0  0.000000        0      0.0       2   object
3        sistema_salud      0  0.000000        0      0.0       7   object
4      nivel_educativo     66  0.018771        0      0.0       7   object
5  ocupacion_jerarquia      0  0.000000        0      0.0       4   object
6         estado_civil      0  0.000000        0      0.0       5    int64
7          ingreso_15k      0  0.000000        0      0.0       2   object

There are 134 NaNs in 'edad' and in 'nivel_educativo'. Also, 'estado_civil' is numerical variable (it has to be categorical).

<h3>Data preparation</h3>

Convert 'estado_civil' from numerical to categorical

In [5]:
data['estado_civil_cat'] = data['estado_civil'].astype('str')

In [7]:
data = data.drop('estado_civil', axis = 1)

In [8]:
status(data)

variable  q_nan     p_nan  q_zeros  p_zeros  unique     type
0                 edad    134  0.038111        0      0.0      70  float64
1                 sexo      0  0.000000        0      0.0       2   object
2             alfabeto      0  0.000000        0      0.0       2   object
3        sistema_salud      0  0.000000        0      0.0       7   object
4      nivel_educativo     66  0.018771        0      0.0       7   object
5  ocupacion_jerarquia      0  0.000000        0      0.0       4   object
6          ingreso_15k      0  0.000000        0      0.0       2   object
7     estado_civil_cat      0  0.000000        0      0.0       5   object

NaNs treament in 'nivel_educativo' (i will add 'nulo' as a new category)

In [9]:
data['nivel_educativo'] = data['nivel_educativo'].fillna(value="nulo") 

In [10]:
status(data['nivel_educativo'])

variable  q_nan  p_nan  q_zeros  p_zeros  unique    type
0  nivel_educativo      0    0.0        0      0.0       8  object

In [12]:
freq_tbl(data['nivel_educativo'])

nivel_educativo  frequency  percentage  \
0                              Secundaria Completa        915    0.260239   
1                                Primaria Completa        673    0.191411   
2                            Secundaria Incompleta        630    0.179181   
3                  Superior Universitaria Completa        596    0.169511   
4                Superior Universitaria Incompleta        419    0.119170   
5  Primaria Incompleta(incluye educación especial)        200    0.056883   
6                                             nulo         66    0.018771   
7                                  Sin instrucción         17    0.004835   

   cumulative_perc  
0         0.260239  
1         0.451650  
2         0.630830  
3         0.800341  
4         0.919511  
5         0.976394  
6         0.995165  
7         1.000000

NaNs treament in 'edad' (first i will discretize the variable, and next, add the 'nulo' category as before)

In [13]:
data['edad_cat'] = pd.qcut(data['edad'], q = 5)

In [14]:
data['edad_cat'] = data['edad_cat'].cat.add_categories('nulo')

In [16]:
freq_tbl(data['edad_cat'])

edad_cat  frequency  percentage  cumulative_perc
0    (32.0, 41.0]        762    0.216724         0.225310
1  (14.999, 32.0]        687    0.195392         0.428445
2    (48.0, 57.0]        673    0.191411         0.627439
3    (57.0, 95.0]        643    0.182878         0.817564
4    (41.0, 48.0]        617    0.175484         1.000000
5            nulo          0    0.000000         1.000000

In [17]:
data['edad_cat']=data['edad_cat'].fillna(value="nulo")

In [18]:
data = data.drop('edad', axis = 1)

In [19]:
status(data)

variable  q_nan  p_nan  q_zeros  p_zeros  unique      type
0                 sexo      0    0.0        0      0.0       2    object
1             alfabeto      0    0.0        0      0.0       2    object
2        sistema_salud      0    0.0        0      0.0       7    object
3      nivel_educativo      0    0.0        0      0.0       8    object
4  ocupacion_jerarquia      0    0.0        0      0.0       4    object
5          ingreso_15k      0    0.0        0      0.0       2    object
6     estado_civil_cat      0    0.0        0      0.0       5    object
7             edad_cat      0    0.0        0      0.0       6  category

The new dataset doesnt have NaNs

Supossing that i will use this data in a predictive model, i will convert all the data to numerical, and save it using Pickle.

In [20]:
data_final = pd.get_dummies(data, drop_first=True)

In [22]:
status(data_final)

variable  q_nan  p_nan  q_zeros  \
0                                          sexo_mujer      0    0.0     2226   
1                                         alfabeto_Si      0    0.0       23   
2   sistema_salud_Mutual/prepaga/servicio de emerg...      0    0.0     3513   
3              sistema_salud_No paga ni le descuentan      0    0.0     1817   
4                               sistema_salud_Ns./Nr.      0    0.0     3512   
5            sistema_salud_Obra social (incluye PAMI)      0    0.0     2043   
6   sistema_salud_Obra social y mutual/prepaga/ser...      0    0.0     3444   
7             sistema_salud_Planes y seguros públicos      0    0.0     3471   
8   nivel_educativo_Primaria Incompleta(incluye ed...      0    0.0     3316   
9                 nivel_educativo_Secundaria Completa      0    0.0     2601   
10              nivel_educativo_Secundaria Incompleta      0    0.0     2886   
11                    nivel_educativo_Sin instrucción      0    0.0     3499   
12    nivel_educativo_Superior Universitaria Completa      0    0.0     2920   
13  nivel_educativo_Superior Universitaria Incompleta      0    0.0     3097   
14                               nivel_educativo_nulo      0    0.0     3450   
15                       ocupacion_jerarquia_Director      0    0.0     3161   
16          ocupacion_jerarquia_Trabajador asalariado      0    0.0     3498   
17                           ocupacion_jerarquia_otro      0    0.0     3506   
18                                     ingreso_15k_si      0    0.0     2542   
19                                 estado_civil_cat_2      0    0.0     3121   
20                                 estado_civil_cat_3      0    0.0     2669   
21                                 estado_civil_cat_4      0    0.0     2550   
22                                 estado_civil_cat_5      0    0.0     3414   
23                              edad_cat_(32.0, 41.0]      0    0.0     2754   
24                              edad_cat_(41.0, 48.0]      0    0.0     2899   
25                              edad_cat_(48.0, 57.0]      0    0.0     2843   
26                              edad_cat_(57.0, 95.0]      0    0.0     2873   
27                                      edad_cat_nulo      0    0.0     3382   

     p_zeros  unique   type  
0   0.633106       2  uint8  
1   0.006542       2  uint8  
2   0.999147       2  uint8  
3   0.516780       2  uint8  
4   0.998862       2  uint8  
5   0.581058       2  uint8  
6   0.979522       2  uint8  
7   0.987201       2  uint8  
8   0.943117       2  uint8  
9   0.739761       2  uint8  
10  0.820819       2  uint8  
11  0.995165       2  uint8  
12  0.830489       2  uint8  
13  0.880830       2  uint8  
14  0.981229       2  uint8  
15  0.899033       2  uint8  
16  0.994881       2  uint8  
17  0.997156       2  uint8  
18  0.722981       2  uint8  
19  0.887656       2  uint8  
20  0.759101       2  uint8  
21  0.725256       2  uint8  
22  0.970990       2  uint8  
23  0.783276       2  uint8  
24  0.824516       2  uint8  
25  0.808589       2  uint8  
26  0.817122       2  uint8  
27  0.961889       2  uint8

In [ ]:
with open('data/d_eph5.pickle', 'wb') as f:
    pickle.dump(data_final, f)